.github/workflows/static.yml

```diff
@@ -37,9 +37,9 @@ jobs:
       - name: Build WebSite
         run: ./.doc/build.sh
       - name: Upload artifact
-        uses: actions/upload-pages-artifact@v1
+        uses: actions/upload-pages-artifact@v3
```

Comment:
Can you pin to the hash instead of the tag? (Tag is modifiable)

Reply:
Fixed.  

The downside to using hashes though is that we don't get bug and security fixes.  The upside of course is we don't get exposed to waterhole attacks.  

It's a tradeoff.  If we are going to use hashes, I think we must actually track these dependencies, so we'll know if something comes up.   A better option might be to just lock down what actions can do, so even if something malicious happens we don't care.

Either way, it's a debate for another day.

Reply:
> The downside to using hashes though is that we don't get bug and security fixes. The upside of course is we don't get exposed to waterhole attacks.

Dependabot should actually bump actions in case of security vulnerabilities (and also keep the hash pinning!):
https://github.com/dependabot/dependabot-core/pull/5951